### DISCUSSION TITLE
Warning about Symmetry boundary face

### DISCUSSION THREAD

**Discussion by pkck28** [mdolab/adflow#242] [Q&A] [answered]
I am running adflow and at the start of the analysis I get a warning:

Symmetry boundary face BC9 of zone domain.00003 is not planar.
Maximum deviation from the mean normal:  7.75206E+00 degrees

I know this is coming from the mesh, but I wanted to know more about this warning and how can I get rid of them. I have following questions about the warning:

- Is mean normal the mean of all the normal of the faces in that particular of zone of the mesh?
- Can I ignore this ignore and not comprise the analysis result?
- How much maximum deviation can be ignored, if ignoring is okay in the first place?

Any help would be appreciated!

**Comment by sseraj** [comment, marked as answer]
The mean normal is the mean of the face normals in that CGNS block _projected in the direction of the symmetry plane normal_. As a result, the mean normal points in the same direction as the normal to the symmetry plane.

The effect of non-planar faces on the accuracy of the analysis depends on the size of the block as well as the angle. For example, I have seen a 2 deg deviation on a small block result in almost no change in the output lift and drag. An 8 deg deviation is most likely too large to ignore.

In addition, if you are planning on running shape optimizations with MACH-Aero, any amount of deviation can be problematic. Even small deviations can cause major issues with the volume mesh warping (IDWarp).

**Reply by pkck28** [reply]
Hi Sabet

Thanks a lot for a reply! I got this mesh by deforming an original wing mesh using IDWarp. I tried default option values and I also tried varying some options values, but that warning is present everytime with small changes in the value of deviation.

So, in gradient based ASO, how to ensure that there are no deviations in mesh at each iteration? I think IDWarp is used to deform the mesh based on the changes in the surface mesh coordinates (correct me if I am wrong). Is there some recommended values (other than default values) for options of USMesh?

Regards,
Pavan.

**Reply by pkck28** [reply]
I was able to get rid of those warnings by setting _useRotations_ options as _False_. Will this have any effect on the analysis result?

Regards,
Pavan.

**Reply by sseraj** [reply]
If you get non-planar warnings after warping, then it is likely that your baseline mesh is not exactly planar. Even deviations around 1e-3 on the baseline mesh can cause bigger problems after warping. You might find the `symmZero` function in [cgnsUtilities](https://github.com/mdolab/cgnsutilities) useful. This will exactly zero the symmetry plane.

`useRotations` mainly affects boundary layer orthogonality, as explained in the [options documentation](https://mdolab-idwarp.readthedocs-hosted.com/en/latest/options.html). Even if using `"useRotations": False` fixes the issue, I would recommend checking the baseline mesh to make sure the symmetry plane is actually planar.

**Reply by pkck28** [reply]
I changed the value of `useRotations` to `True` in USMesh. Then, I used `symmZero` on the base mesh, but the warning was still present - I think base mesh is good. So, I tried `symmZero` on the deformed mesh and there were no warnings. This way of dealing with those warnings should be fine, right?

**Reply by sseraj** [reply]
`symmZero` is not meant to be used inside the optimization loop but should be okay for analysis. If you are running into this issue repeatedly in an optimization, then maybe just set `useRotations` to `False`.

**Reply by pkck28** [reply]
I am just running analysis, it is not within an optimization loop. Thanks a lot, sabet!
